MyOnboarding/src/MyOnboardingApp.Api/Controllers/TodoListController.cs

```diff
@@ -15,45 +16,206 @@ namespace MyOnboardingApp.Api.Controllers
-            => Ok(await _repository.DeleteItemAsync(id));
+        public async Task<IHttpActionResult> DeleteAsync(Guid id)
+        {
+            ValidateNonEmptyGuid(id);
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var deletedItemWithStatus = await _deleteService.DeleteItemAsync(id);
+            if (!deletedItemWithStatus.WasOperationSuccessful)
+            {
+                return StatusCode(HttpStatusCode.NoContent);
+            }
+
+            return Ok(deletedItemWithStatus.Item);
+        }
+
+
+        private static TodoListItem PrepareItemForCreation(TodoListItem item)
+        {
+            item.Id = Guid.Empty;
+            return item;
+        }
+
+
+        private TodoListItem CompleteItemBeforePut(TodoListItem replacingItem, TodoListItem existingItem)
+        {
+            replacingItem.CreationTime = existingItem.CreationTime;
+            replacingItem.LastUpdateTime = existingItem.LastUpdateTime;
+            return replacingItem;
+        }
+
+
+        private void ValidateNonEmptyGuid(Guid id)
+        {
+            if (id == Guid.Empty)
+            {
+                ModelState.AddModelError(nameof(id), "Id must not be empty.");
+            }
+        }
+
+
+        private void ValidateIdPost(Guid id)
+        {
+            if (id != Guid.Empty)
+            {
+                ModelState.AddModelError(nameof(id), "Id must be empty.");
+            }
+        }
+
+
+        private void ValidateTimePost(DateTime time, string field)
+        {
+            if (time != DateTime.MinValue)
+            {
+                ModelState.AddModelError(field, "Time is supposed to be DateTime.MinValue.");
+            }
+        }
+
+
+        private void ValidateNotEmptyText(string text)
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                ModelState.AddModelError(nameof(text), "Text must not be empty.");
+            }
+        }
+
+
+        private void ValidateItemBeforePost(TodoListItem item)
+        {
+            if (item == null)
+            {
+                ModelState.AddModelError(nameof(item), "Item from the request body must not be null.");
+                return;
+            }
+
+            ValidateNotEmptyText(item.Text);
+            ValidateTimePost(item.CreationTime, "CreationTime");
+            ValidateTimePost(item.LastUpdateTime, "LastUpdateTime");
+            ValidateIdPost(item.Id);
+        }
+
+
+        private void ValidateItemBeforePut(TodoListItem item, Guid idFromUrl)
+        {
+            if (item == null)
+            {
+                ModelState.AddModelError(nameof(item), "Item from the request body must not be null.");
+                return;
+            }
+
+            if (item.Id == Guid.Empty)
+            {
+                ModelState.AddModelError(nameof(item.Id), "Id must not be empty.");
```

Comment:
```suggestion
                ModelState.AddModelError(nameof(TodoListItem.Id), "Id must not be empty.");
```
This is a total nuance, however, I personally believe that you refer to a property defined in a type rather than to property defined on a variable (especially since this is not a TypeScript habitat 👼)